app/src/main/java/ar/com/wolox/android/example/ui/example/fragment/NewDetailFragment.kt

```diff
@@ -0,0 +1,152 @@
+    }
+
+    //Cargar detalles de la noticia
+    override fun loadData(new : News){
+        with(binding){
+            //Guardo el new
+            currentNew = new
+
+            //Convertir la fecha string a date
+            val strdate = new.date.convertToDate()
+
+            title.text = new.commenter
+            description.text = new.comment
+            date.text = prettyTime.format(strdate?.toDate())
+
+            //Compruebo si el id del usuario esta en el array del like
+            if(new.likes.contains(presenter.getUserId())) {
+                like.setImageResource(R.mipmap.ic_like_on_large)
+            }
+
+            //Cargar imagen en el tabbar y del fullscreen
+            val imageBar = (activity as? NewDetailActivity)?.getImageBar()
+            val imagefull = (activity as? NewDetailActivity)?.getImageFull()
+            if(imageBar != null && imagefull !=null) {
+                Glide.with(requireContext())
+                    .load(new.avatar)
+                    .into(imageBar)
+
+                Glide.with(requireContext())
+                    .load(new.avatar)
+                    .into(imagefull)
+
+                //Si se realiza click sobre la imagen del tabbar, se visualiza el fullscreen
+                imageBar.setOnClickListener(){
+                    imagefull.visibility = VISIBLE
+                }
+
+                //Si se realiza click sobre el fullscreen, se oculta el fullscreen
+                imagefull.setOnClickListener(){
+                    imagefull.visibility = GONE
+                }
+            }
+        }
+    }
+
+    override fun setListeners() {
+        with(binding){
+            like.setOnClickListener{
+                presenter.onLikeClicked(currentNew.id)
+            }
+            swipe.setOnRefreshListener {
+                presenter.refreshNew(currentNew.id)
+            }
+        }
+    }
+
+    //Para habilitar o desahabilitar la imagen del like
+    override fun enabledLike(enabled : Boolean){
+        with(binding){
+            like.isEnabled = enabled
+        }
+    }
+
+    //Para cambiar el estado del like
+    override fun changeLike() {
+        with(binding){
+            val userid = presenter.getUserId()
+
+            //Si el array del like contiene el id del usuario, cambio el icono del like al rellenito
+            //sino, cambio al vacio
+            if(currentNew.likes.contains(userid)) {
+                currentNew.likes.remove(userid)
+                like.setImageResource(R.mipmap.ic_like_off_large)
+            }
+            else {
+                currentNew.likes.add(userid)
+                like.setImageResource(R.mipmap.ic_like_on_large)
+            }
+
+            //Para guardar que algo cambio y refrescar las noticias
+            context?.getSharedPreferences(BaseConfiguration.SHARED_PREFERENCES_NAME,MODE_PRIVATE)?.edit()?.putBoolean(Extras.Constantes.NEW_CHANGE,true)?.apply()
```

Comment:
You shouldn't write data from view (Fragments or Activities). I recommend that you do not use a flag and call from the NewsFragment onResume again to the presenter function that calls the endpoint to fetch data.